x-pack/plugins/security/server/feature_usage/feature_usage_service.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Optional } from '@kbn/utility-types';
+import { FeatureUsageServiceSetup, FeatureUsageServiceStart } from '../../../licensing/server';
+import { RoleKibanaPrivilege } from '../../common/model';
+
+interface SetupDeps {
+  featureUsage: FeatureUsageServiceSetup;
+}
+
+interface StartDeps {
+  featureUsage: FeatureUsageServiceStart;
+}
+
+export interface SecurityFeatureUsageServiceStart {
+  recordPreAccessAgreementUsage: () => void;
+  recordSubFeaturePrivilegeUsage: () => void;
+}
+
+export class SecurityFeatureUsageService {
+  public setup({ featureUsage }: SetupDeps) {
+    featureUsage.register('security_sub_feature_privileges');
+    featureUsage.register('security_pre_access_agreement');
+  }
+
+  public start({ featureUsage }: StartDeps): SecurityFeatureUsageServiceStart {
```

Comment:
This `SecurityFeaureUsageService` feels rather unnecessary given the amount of work it's doing. It's really just a very thin facade over the `featureUsage` service exposed by the licensing plugin.

I opted to do this anyway because:
1) We've previously tried to isolate the security plugin from the licensing plugin by introducing the `SecurityLicense` service, so it felt appropriate to continue this initiative.
2) We expect the `featureUsage` service to change in the near future, but we don't yet know what that will look like. Having this encapsulated here will hopefully prevent sprawling changes to the security plugin when this happens.